### PR TITLE
Fuzzing - added support for allOf, anyOf, oneOf schemas

### PR DESCRIPTION
**Pull request by thim81**

### CODE REVIEW

**Comment by thim81** [comment]
linked to #335

File: src/application/Fuzzer.ts [hunk 1]
```diff
@@ -764,7 +764,7 @@ export class Fuzzer {
   }
 
   public analyzeFuzzJsonSchema(
-    jsonSchema: OpenAPIV3.SchemaObject | undefined
+    jsonOrgSchema: OpenAPIV3.SchemaObject | undefined
```
[the remark this answers is not in the document]
Comment (thim81) [reply src/application/Fuzzer.ts]: Yes, Good suggestion. I'll correct it.

File: src/application/Fuzzer.ts [hunk 2]
```diff
@@ -775,9 +775,46 @@ export class Fuzzer {
...
+    // Copy jsonSchema to keep the original jsonSchema untouched
+    const jsonSchema = { ...jsonOrgSchema } as OpenAPIV3.SchemaObject
 
-    const skipSchemaKeys = ['properties', 'items']
+    // Handle allOf properties
+    if (jsonSchema.allOf) {
+      // Merge allOf properties
+      jsonSchema.allOf.forEach(function (s) {
+        if ('properties' in s) {
+          jsonSchema.properties = Object.assign(jsonSchema.properties || {}, s.properties || {})
```
[the remark this answers is not in the document]
Comment (thim81) [reply src/application/Fuzzer.ts:787]: Indeed, the IF has that covered.

[the remark this answers is not in the document]
Comment (nicklloyd) [reply src/application/Fuzzer.ts:787]: as you're already checking `if ('properties' in s)`

**Pull request merged**